src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -611,27 +683,119 @@ def getJobOutput(self, jobID, _localDir=None):
 
         # Submit the GET request to retrieve outputs
+        self.log.debug("Retrieving pilot logging info for %s" % pilotRef)
         response = self.session.get(query, headers=self.headers, timeout=self.arcRESTTimeout)
         if not response.ok:
-            self.log.error("Error downloading stdout", "for %s: %s" % (job, response.text))
-            return S_ERROR("Failed to retrieve at least some output for %s" % jobID)
-        output = response.text
+            self.log.error(
+                "Error downloading logging info", "for %s: %s %s" % (job, response.status_code, response.reason)
+            )
+            return S_ERROR("Failed to retrieve logging info for %s" % jobID)
+        loggingInfo = response.text
+
+        return S_OK(loggingInfo)
 
-        query = self._UrlJoin([command, job, "session", stamp, ".err"])
+    #############################################################################
+
+    def _getListOfAvailableOutputs(self, pilotRef, job):
+        """Request a list of outputs available for a given jobID.
+
+        :param str pilotRef: DIRAC job ID
+        :param str job: AREX job ID
+        :return list: names of the available outputs
+        """
+        command = "jobs/"
+        query = self._UrlJoin([command, job, "session"])
         if query.startswith("Unknown"):
             return S_ERROR("Problem creating REST query %s" % query)
+
+        # Submit the GET request to retrieve the names of the outputs
+        self.log.debug("Retrieving the names of the pilot outputs for %s" % pilotRef)
         response = self.session.get(query, headers=self.headers, timeout=self.arcRESTTimeout)
+
         if not response.ok:
-            self.log.error("Error downloading stderr", "for %s: %s" % (job, response.text))
-            return S_ERROR("Failed to retrieve at least some output for %s" % jobID)
-        error = response.text
+            self.log.error(
+                "Error getting available outputs", "for %s: %s %s" % (pilotRef, response.status_code, response.reason)
+            )
+            return S_ERROR("Failed to retrieve at least some outputs for %s" % pilotRef)
+
+        if not response.text:
+            return S_ERROR("There is no output for job %s" % pilotRef)
+
+        return S_OK(response.json()["file"])
+
+    def getJobOutput(self, jobID, workingDirectory=None):
+        """Get the specified job standard output and error files.
+        The output is returned as strings.
+        """
+        if not self.session:
+            return S_ERROR("REST interface not initialised. Cannot get job output.")
 
-        return S_OK((output, error))
+        # Get a proxy
+        result = self._prepareProxy()
+        if not result["OK"]:
+            self.log.error("AREXComputingElement: failed to set up proxy", result["Message"])
+            return result
+        self.session.cert = Locations.getProxyLocation()
+
+        # Extract stamp from the Job ID
+        if ":::" in jobID:
```

Comment:
Not sure how much connected, but beware of what's being done by @atsareg  with https://github.com/DIRACGrid/DIRAC/pull/6405.

Reply:
From what I understand, it is ok.